packages/diagrams/lib/src/stream_builder.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:diagram_capture/diagram_capture.dart';
+import 'package:flutter/material.dart';
+
+import 'diagram_step.dart';
+
+const String _streamBuilder = 'stream_builder';
+const String _streamBuilderError = 'stream_builder_error';
+const Duration _pauseDuration = Duration(seconds: 1);
+final Duration _totalDuration = _pauseDuration * 4;
+
+class StreamBuilderDiagram extends StatefulWidget implements DiagramMetadata {
```

Comment:
These are really neat! 👍  Let's add some color or icons like with the Future animations you made previously.

Reply:
Added some iconography and animations!